### PR TITLE
Add implementation for binary ops.

### PR DESCRIPTION
**Pull request by BruceDai**
The binary tests were ported form webnn-polyfill.
Should we update expected data of double precision which then would benefit to WPT tests?
 
@huningxin PTAL, thanks.

### CODE REVIEW

File: src/binary.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+/**
+ * Compute the element-wise binary operation of the two input tensors and given operation name.
+ * @param {Tensor} inputA
+ * @param {Tensor} inputB
+ * @param {String} name 'add'|'sub'|'mul'|'div'|'max'|'min'|'pow'
+ * @return {Tensor}
+ */
+export function binary(inputA, inputB, name) {
```
Comment (huningxin) [inline src/binary.js:50]: The third parameter probably could be a function object that computes the binary.

Comment (huningxin) [reply src/binary.js:50]: I prefer to having individual function for each binary op, such as `add`, `sub`. You can just call the `binary` with a corresponding function object.

File: src/binary.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  const outputShape = getBroadcastShape(inputA.shape, inputB.shape);
```
Comment (huningxin) [inline src/binary.js:51]: You could broadcast input a and b by PR #2, PTAL.

File: src/common.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  return shape.reduce((accumulator, currentValue) => accumulator * currentValue, 1);
+}
+
+/**
+ * Get broadcast shape of given two input shapes, throw error if they're incompatible.
+ * @param {Array} shapeA
+ * @param {Array} shapeB
+ * @return {Array}
+ */
+export function getBroadcastShape(shapeA, shapeB) {
```
Comment (huningxin) [inline src/common.js:18]: Probably we don't need this, please take a look at #2.

File: src/tensor.js [hunk 4]
```diff
@@ -77,4 +79,23 @@ export class Tensor {
...
     return this.data[this.indexFromLocation(location)];
   }
+
+  /**
+   * Get value from boradcastd tensor of `broadcastShape` by `broadcastLocation`.
+   * @param {Number} broadcastLocation
+   * @param {Array} broadcastShape
+   * @return {Number}
+   */
+  getValueBroadcast(broadcastLocation, broadcastShape) {
```
Comment (huningxin) [inline src/tensor.js:89]: Probably we don't need this one, PTAL #2.

File: src/common.js (hunk 3 above)
Comment (huningxin) [inline src/common.js:18]: If this is only used by binary, please move it there.

File: src/common.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+/**
+ * Get a product number for a given shape array.
+ * @param {Array} shape
+ * @return {Number}
+ */
+export function product(shape) {
```
Comment (huningxin) [inline src/common.js:8]: sizeOfShape?

File: src/binary.js (hunk 1 above)
Comment (BruceDai) [reply src/binary.js:50]: Thanks your suggestion, I will update it.

**Comment by BruceDai** [comment]
Thanks @huningxin, I will update this PR based on PR #2.

File: src/binary.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Compute the element-wise binary operation of two input tensors.
+ * @param {Tensor} inputA
+ * @param {Tensor} inputB
+ * @param {Function} binaryFunc
+ * @return {Tensor}
+ */
+function binary(inputA, inputB, binaryFunc) {
+  const outputShape = getBroadcastShape(inputA.shape, inputB.shape);
+  const inputABroadcast = broadcast(inputA, outputShape);
```
Comment (BruceDai) [inline src/binary.js:15]: @huningxin Should we consider optimizing of memory usage here?

**Review by BruceDai — COMMENTED** [review]
@huningxin I've updated PR based on PR #2, please take a review again, thanks.

File: src/binary.js (hunk 6 above)
Comment (huningxin) [reply src/binary.js:15]: Thanks for this consideration. Code readability and algorithm simplicity are focus of the baseline implementation. If we see issues later, we can optimize.

**Review by huningxin — APPROVED** [review]
LGTM, thanks @BruceDai

**Pull request merged**